pkg/maintenance/maintenance.go

```diff
@@ -0,0 +1,27 @@
+package maintenance
+
+import (
+	"sigs.k8s.io/controller-runtime"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type Maintenance interface {
+	Start() error
+	End() error
+}
+
+func NewMaintenace() (Maintenance, error) {
+	// new client to avoid caching issues
```

Comment:
Actually @wanghaoran1988 why do I have to do this?
I need a non-cached client to get this route. any ideas?

Reply:
Could you point me why we need the non-cached client?

Reply:
The cached client wasn't retrieving the alertmanager route.
I switched to the non-cached and it was retrieved. I will see if the cached works once I merged into your branch but was wondering if you had seen this

Reply:
@sedroche How you deploy the operator, if it's by a deployment objet, the following must be set:
```
 - name: WATCH_NAMESPACE
   value: ""
```
or you test locally with:
`operator-sdk run --local --verbose --watch-namespace ""`

Reply:
Just locally from goland. Have that set in my environemnt too. weird.